From 81f61eb4966e4652382ebee32f1830cfb4ca9126 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Mon, 8 May 2017 12:40:00 -0400
Subject: manager: stop greeter explicitly when finishing display

If GDM shuts down while the login screen is active, we fail to
kill off the login screen session.

This commit fixes that, by explicitly stopping the greeter session
on the display (if there is one)

https://bugzilla.gnome.org/show_bug.cgi?id=780213
---
 daemon/gdm-manager.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/daemon/gdm-manager.c b/daemon/gdm-manager.c
index 994886f..d080b30 100644
--- a/daemon/gdm-manager.c
+++ b/daemon/gdm-manager.c
@@ -2736,6 +2736,7 @@ finish_display (const char *id,
                 GdmDisplay *display,
                 GdmManager *manager)
 {
+        gdm_display_stop_greeter_session (display);
         if (gdm_display_get_status (display) == GDM_DISPLAY_MANAGED)
                 gdm_display_unmanage (display);
         gdm_display_finish (display);
-- 
cgit v0.12

